src/RobotSimulator.elm

```diff
@@ -0,0 +1,305 @@
+    | Interval
+
+
+addVectors : Vector -> Vector -> Vector
+addVectors baseVector addedVector =
+    Vector (baseVector.x + addedVector.x) (baseVector.y + addedVector.y)
+
+
+asShotsIn : Model -> List GameObject -> Model
+asShotsIn model newShots =
+    { model | shots = newShots }
+
+
+asRobotIn : Model -> GameObject -> Model
+asRobotIn model newRobot =
+    { model | robot = newRobot }
+
+
+asPositionIn : GameObject -> Vector -> GameObject
+asPositionIn gameObject newPosition =
+    { gameObject | position = newPosition }
+
+
+asDirectionIn : GameObject -> Direction -> GameObject
+asDirectionIn gameObject newDirection =
+    { gameObject | direction = newDirection }
+
+
+
+-- Init
+
+
+init : () -> ( Model, Cmd Msg )
+init _ =
+    let
+        robot =
+            GameObject { x = 50, y = 50 } { x = 5, y = 5 } Up
+
+        shots =
+            []
+    in
+    ( Model robot shots, Cmd.none )
+
+
+
+-- Update
+
+
+inBorders : GameObject -> Bool
+inBorders gameObject =
+    let
+        position =
+            gameObject.position
+    in
+    position.x < screenWidth && position.x > 0 && position.y < screenHeight && position.y > 0
+
+
+moveByVelocity : GameObject -> GameObject
+moveByVelocity gameObject =
+    { gameObject | position = addVectors gameObject.position gameObject.velocity }
+
+
+moveShots : List GameObject -> List GameObject
+moveShots shots =
+    List.map moveByVelocity <| List.filter inBorders <| shots
+
+
+directionToVelocity : Direction -> Vector -> Vector
+directionToVelocity dir velocity =
+    case dir of
+        Right ->
+            Vector velocity.x 0
+
+        Left ->
+            Vector -velocity.x 0
+
+        Up ->
+            Vector 0 -velocity.y
+
+        Down ->
+            Vector 0 velocity.y
+
+
+moveRobot : Direction -> GameObject -> GameObject
+moveRobot dir robot =
+    let
+        position =
+            robot.position
+    in
+    asPositionIn robot <|
+        addVectors position <|
+            directionToVelocity dir robot.velocity
+
+
+formatModelOutput : Model -> ( Model, Cmd Msg )
+formatModelOutput model =
+    ( model, Cmd.none )
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    formatModelOutput <|
+        case msg of
+            Move dir ->
+                asRobotIn model <|
+                    let
+                        movedRobot =
+                            moveRobot dir <|
+                                asDirectionIn model.robot dir
+                    in
+                    if inBorders movedRobot then
+                        movedRobot
+
+                    else
+                        model.robot
+
+            Shoot ->
+                let
+                    direction =
+                        model.robot.direction
+
+                    robotPosition =
+                        model.robot.position
+                in
+                asShotsIn model <|
+                    GameObject (Vector (robotPosition.x + 7) (robotPosition.y + 7))
```

Comment:
Where are these `7`s coming from?
Are they supposed to be the `velocity` calculated by `directionToVelocity direction shotVelocity`?
If they're some global settings, they should be another parameter for update (this way we easily partially apply settings when the module is used from another module (or right before `main` of some sort)).